pkg/services/ngalert/api/api_ruler.go

```diff
@@ -347,12 +381,51 @@ func (srv RulerSrv) updateAlertRulesInGroup(c *models.ReqContext, namespace *mod
 		}
 
 		if groupChanges.isEmpty() {
-			authorizedChanges = groupChanges
+			mergedChanges = groupChanges
 			logger.Info("no changes detected in the request. Do nothing")
 			return nil
 		}
 
-		authorizedChanges, err = authorizeRuleChanges(namespace, groupChanges, func(evaluator accesscontrol.Evaluator) bool {
+		provenances, err := srv.provenanceStore.GetProvenances(c.Req.Context(), c.OrgId, (&ngmodels.AlertRule{}).ResourceType())
+		if err != nil {
+			return err
+		}
+
+		provisioningChanges := &changes{}
+		// New rules don't need to be checked for provenance, just copy the whole slice.
+		provisioningChanges.New = groupChanges.New
+		for _, rule := range groupChanges.Update {
+			if provenance, exists := provenances[rule.Existing.UID]; (exists && provenance == ngmodels.ProvenanceNone) || !exists {
+				provisioningChanges.Update = append(provisioningChanges.Update, rule)
+			}
+		}
+		for _, rule := range groupChanges.Delete {
+			if provenance, exists := provenances[rule.UID]; (exists && provenance == ngmodels.ProvenanceNone) || !exists {
+				provisioningChanges.Delete = append(provisioningChanges.Delete, rule)
+			}
+		}
+
+		if provisioningChanges.isEmpty() {
+			logger.Info("no changes detected that have 'none' provenance in the request. Do nothing",
+				"provenance_invalid_add", len(groupChanges.New),
+				"provenance_invalid_update", len(groupChanges.Update),
+				"provenance_invalid_delete", len(groupChanges.Delete))
+			return nil
+		}
+
+		if len(groupChanges.Delete) > len(provisioningChanges.Delete) {
+			logger.Info("provenance is not 'none' for one or many rules in the group that should be deleted. those rules will be skipped",
+				"expected", len(groupChanges.Delete),
+				"allowed", len(provisioningChanges.Delete))
+		}
+
+		if len(groupChanges.Update) > len(provisioningChanges.Update) {
+			logger.Info("provenance is not 'none' for one or many rules in the group that should be updated. those rules will be skipped",
+				"expected", len(groupChanges.Update),
+				"allowed", len(provisioningChanges.Update))
+		}
```

Comment:
I think this should go after authz logic and check `authorizedChanges`

Reply:
So what you suggest is to iterate over the result of `authorizeRuleChanges` and remove the rules that are provisioned, right? How would that prevent the logs? A rule a user could read could be provisioned.

Reply:
I think I got it, we would emit two logs for one invalid rule - once for auth, once for alerting. By changing the order we would only emit one for auth as in the provenance check it would already be removed.

Reply:
I changed it 👍